packages/editor/src/browser/navigation/navigation-location-service.ts

```diff
@@ -233,4 +235,45 @@ ${this.stack.map((location, i) => `${i}: ${JSON.stringify(NavigationLocation.toO
 ----- o -----`;
     }
 
+    // A history of recently closed editors.
+    private recentlyClosedEditors: EditorWidget[] = [];
+    private static readonly MAX_RECENTLY_CLOSED_EDITORS = 30;
+
+    /**
+     * Appends new closed editor to the history.
+     * Removes old editor that is duplicated of the given editor if exists.
+     * Removes old editors if the number of editors in the history exceeds MAX_RECENTLY_CLOSED_EDITORS.
```

Comment:
Keep the description simple for callers, in the body of the method you can describe the method further.